src/indexer/index_sorter.rs

```diff
@@ -0,0 +1,426 @@
+use super::SegmentWriter;
+use crate::{DocId, IndexSettings, Order, TantivyError};
+use std::cmp::Reverse;
+
+/// Struct to provide mapping from old docid to new docid and vice versa
+pub struct DocidMapping {
+    new_docid_to_old: Vec<u32>,
+    old_docid_to_new: Vec<u32>,
+}
+
+impl DocidMapping {
+    /// returns the new docid for the old docid
+    pub fn get_new_docid(&self, docid: DocId) -> DocId {
+        self.old_docid_to_new[docid as usize]
+    }
+    /// returns the old docid for the new docid
+    pub fn get_old_docid(&self, docid: DocId) -> DocId {
+        self.new_docid_to_old[docid as usize]
+    }
+    /// iterate over old docids in order of the new docids
+    pub fn iter_old_docids(&self) -> std::slice::Iter<'_, u32> {
+        self.new_docid_to_old.iter()
+    }
+}
+
+pub(crate) fn sort_index(
```

Comment:
shouldn;'t we remove this function and call get_doc_id_mapping directly?

Reply:
yes, this is some residue when I started